resources.go

```diff
@@ -1501,3 +1502,84 @@ func Provider() tfbridge.ProviderInfo {
 
 	return prov
 }
+
+func renameLegacyModules(prov *tfbridge.ProviderInfo) {
+
+	renameResourceWithAlias := func(
+		tfName string,
+		tokName string,
+		newTokName string,
+		legacyModule string,
+		currentModule string,
+		info *tfbridge.ResourceInfo) {
+
+		legacyTFName := tfName + "_legacy"
+
+		if info == nil {
+			info = &tfbridge.ResourceInfo{}
+		}
+		legacyInfo := *info
+		currentInfo := *info
+
+		legacyInfo.Tok = gcpResource(legacyModule, tokName)
+		legacyType := legacyInfo.Tok.String()
+
+		if newTokName != "" {
+			tokName = newTokName
+		}
+
+		currentInfo.Tok = gcpResource(currentModule, tokName)
+		currentInfo.Aliases = []tfbridge.AliasInfo{
+			{Type: &legacyType},
+		}
+
+		if legacyInfo.Docs == nil {
+			legacyInfo.Docs = &tfbridge.DocInfo{
+				Source: tfName[len("azurerm_"):] + ".html.markdown",
+			}
+		}
+
+		prov.Resources[tfName] = &currentInfo
+		prov.Resources[legacyTFName] = &legacyInfo
+		prov.P.ResourcesMap[legacyTFName] = prov.P.ResourcesMap[tfName]
+	}
+
+	//renameDataSourceWithAlias := func(
+	//	tfName string,
+	//	tokName string,
+	//	newTokName string,
+	//	legacyModule string,
+	//	currentModule string,
+	//	info *tfbridge.DataSourceInfo) {
+	//
+	//	legacyTFName := tfName + "_legacy"
+	//
+	//	if info == nil {
+	//		info = &tfbridge.DataSourceInfo{}
+	//	}
+	//	legacyInfo := *info
+	//	currentInfo := *info
+	//
+	//	legacyInfo.Tok = gcpDataSource(legacyModule, tokName)
+	//
+	//	if newTokName != "" {
+	//		tokName = newTokName
+	//	}
+	//
+	//	currentInfo.Tok = gcpDataSource(currentModule, tokName)
+	//
+	//	if legacyInfo.Docs == nil {
+	//		legacyInfo.Docs = &tfbridge.DocInfo{
+	//			Source: tfName[len("azurerm_"):] + ".html.markdown",
+	//		}
+	//	}
+	//
+	//	prov.DataSources[tfName] = &currentInfo
+	//	prov.DataSources[legacyTFName] = &legacyInfo
+	//	prov.P.DataSourcesMap[legacyTFName] = prov.P.DataSourcesMap[tfName]
+	//}
```

Comment:
Should we delete these lines?

Reply:
I added these specifically because of that's the func to rename DataSources. I am going to investigate moving this into `pulumi-terraform-bridge` at some point. But as of now, it's not hugely understandable how we do this

Reply:
I referred to the commented out lines

Reply:
Yeah - we have 2 funcs, 1 to rename resources (which is used) and 1 to rename DataSources, which is what you are referring to

This is only there for now until we can move this back to tfprovider - it will disappear soon:tm: